### PR TITLE
test: added passed cases for init_by_lua*.

### PR DESCRIPTION
**Pull request by doujiang24**
Now we can raise an error in init_by_lua*, which will make the HUP reload failed with nothing changed.
I just added some test case to cover it.
With these test cases, this pr(https://github.com/openresty/lua-nginx-module/pull/974) will failed:
https://travis-ci.org/doujiang24/lua-nginx-module/builds/206939408
(Because nginx think init_module should not failed: https://github.com/nginx/nginx/blob/master/src/core/ngx_cycle.c#L629)

I hereby granted the copyright of the changes in this pull request
to the authors of this lua-nginx-module project.

### CODE REVIEW

**Comment by agentzh** [comment]
@doujiang24 That's interesting. Will you also test the case where `lua_shared_dict` is used?

**Comment by doujiang24** [comment]
@agentzh Sure, added some more test cases :)

File: .travis.yml [hunk 1]
```diff
@@ -65,7 +65,7 @@ install:
   - if [ ! -f download-cache/drizzle7-$DRIZZLE_VER.tar.gz ]; then wget -P download-cache http://openresty.org/download/drizzle7-$DRIZZLE_VER.tar.gz; fi
   - if [ ! -f download-cache/pcre-$PCRE_VER.tar.gz ]; then wget -P download-cache http://ftp.cs.stanford.edu/pub/exim/pcre/pcre-$PCRE_VER.tar.gz; fi
   - if [ ! -f download-cache/openssl-$OPENSSL_VER.tar.gz ]; then wget -P download-cache https://www.openssl.org/source/openssl-$OPENSSL_VER.tar.gz; fi
-  - git clone https://github.com/openresty/test-nginx.git
+  - git clone -b skip https://github.com/doujiang24/test-nginx.git
```
Comment (agentzh) [inline .travis.yml]: Is it still required to use your own `test-nginx` branch for this?

Comment (doujiang24) [inline .travis.yml]: @agentzh yes, this PR haven't be merged yet, https://github.com/openresty/test-nginx/pull/59

**Comment by agentzh** [comment]
Merged. Thanks!

**Pull request closed without merging**